codemods/techTile.js

```diff
@@ -0,0 +1,113 @@
+const visit = require('unist-util-visit');
+const path = require('path');
+const {
+  findAttribute,
+  isMdxBlockElement,
+  hasClassName,
+} = require('./utils/mdxast');
+const {
+  mdxAttribute,
+  mdxValueExpression,
+  mdxBlockElement,
+  mdxSpanElement,
+} = require('./utils/mdxast-builder');
+const { root } = require('mdast-builder');
+const stringify = require('./utils/mdxast-stringify');
+const toString = require('mdast-util-to-string');
+const { select } = require('unist-util-select');
+const { camelCase } = require('lodash');
+
+const isTechTile = isMdxBlockElement('TechTile');
+
+const techTile = () => (tree) => {
+  const imageImports = [];
+
+  visit(
+    tree,
+    (node) => isMdxBlockElement('a', node) && hasClassName('tech-tile', node),
+    (node) => {
+      node.name = 'TechTile';
+
+      const href = findAttribute('href', node);
+      const image = select('image', node.children[0]);
+      const paragraph = node.children[1];
+
+      const isRelativeImport = Boolean(image.url.match(/^\.\.?\//));
+      const importName = camelCase(
+        path.basename(image.url, path.extname(image.url))
+      );
+
+      if (isRelativeImport) {
+        imageImports.push({ name: importName, path: image.url });
+      }
+
+      node.attributes = [
+        mdxAttribute('name', toString(paragraph)),
+        href && mdxAttribute('to', href),
+        mdxAttribute(
+          'icon',
+          mdxValueExpression(
+            stringify(
+              root([
+                mdxSpanElement(
+                  'img',
+                  [
+                    mdxAttribute(
+                      'src',
+                      isRelativeImport
+                        ? mdxValueExpression(importName)
+                        : image.url
+                    ),
+                    image.title && mdxAttribute('title', image.title),
+                    image.alt && mdxAttribute('alt', image.alt),
+                  ].filter(Boolean)
+                ),
+              ])
+            ).trim()
+          )
+        ),
+      ].filter(Boolean);
+
+      node.children = [];
```

Comment:
Since my mind is on localization, we should note that we also need to add a handler for the techTile component. I wonder if there's a way to easily always add a new component to handled with the serialize, deserialize function (in case we forget to add it the handlers for localization).

Reply:
Good catch! I'll update this to include a deserializer.

> I wonder if there's a way to easily always add a new component to handled with the serialize, deserialize function

We talked about this early on and opted for it to be loud when it fails. The reason is because components may need to do some special logic other than the default, and we didn't want to miss something in the localization if we introduced a new component and forgot to add a handler for it. In this case, its better to fail than to be partially wrong.

For example, lets say we introduced the following component:

```
<Dropdown title="Pick something">
  <Dropdown.Item label="Option 1" />
  <Dropdown.Item label="Option 1" />
</Dropdown>
```

If we decided to use default serialize/deserializers, this would mean that none of these props would get serialized in a way that translators would be able to use (because we base64 encode all props). In this case, we'd get back a translated doc with untranslated dropdown items.

By failing, we are able to catch this much easier than if we had decided to use a default serialize/deserialize.